source/common/unified_memory_allocation/include/uma/base.h

```diff
@@ -0,0 +1,59 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef UMA_BASE_H
+#define UMA_BASE_H 1
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define UMA_VERSION_MAJOR ((uint64_t)0)
+#define UMA_VERSION_MINOR ((uint64_t)9)
+
+#define UMA_VERSION_ENCODE(major, minor) (((uint64_t) major << 32) | (uint64_t) minor)
```

Comment:
I think the user should not be responsible for passing version numbers to this macro. This macro should return the version based on the values specified in this header.

Reply:
Yeah, that was my initial implementation but I'm not sure. In some cases, it might be useful to encode/decode particular (not necessarily newest) version. It's similar to how it's done for UR: https://github.com/oneapi-src/unified-runtime/blob/main/include/ur_api.h#L4006

Perhaps we should just have those generic macros to decode/encode + something like 'UMA_VERSION_CURRENT'?

Reply:
Done. I've also make the naming of VERSION macros consistent with the ones in UR.